src/js/components/RadioButtonGroup/__tests__/RadioButtonGroup-test.tsx

```diff
@@ -200,74 +199,88 @@ describe('RadioButtonGroup', () => {
   });
 
   test('Works with keyboard', async () => {
+    const radioGroupOptions = [
+      {
+        id: 'ONE',
+        value: '1',
+        label: 'radio button 1',
+      },
+      {
+        id: 'TWO',
+        value: '2',
+        label: 'radio button 2',
+      },
+      {
+        id: 'THREE',
+        value: '3',
+        label: 'radio button 3',
+      },
+    ];
+
     const onChange = jest.fn();
 
-    const { getByTestId } = render(
+    const { getByRole } = render(
```

Comment:
```javascript
render(
      <Grommet>
        <RadioButtonGroup
          name="test"
          value="2"
          options={radioGroupOptions}
          onChange={onChange}
        />
      </Grommet>,
    );

    // Focus radio '2' button and simulate ArrowDown key
    // should result in selecting radio '3'
    const middleRadioBtn = screen.getByRole('radio', { name: 'radio button 2' });
    middleRadioBtn.focus();
    expect(middleRadioBtn).toHaveFocus();
    await waitFor(() => expect(middleRadioBtn).toHaveFocus(), {
      timeout: 200,
    });

    // focusing the radio button results in internal state update
    // so we wait (`act`) after focusing
    userEvent.type(middleRadioBtn, '{arrowDown}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 3' }),
      }),
    );

    // Focus radio '2' button and simulate ArrowUp key
    // should result in selecting radio '1'
    middleRadioBtn.focus();
    await waitFor(() => expect(middleRadioBtn).toHaveFocus(), {
      timeout: 200,
    });

    userEvent.type(middleRadioBtn, '{arrowUp}');

    expect(onChange).toHaveBeenLastCalledWith(
      expect.objectContaining({
        target: screen.getByRole('radio', { name: 'radio button 1' }),
      }),
    );
```

I think we can change the test to something like this. For what I can tell it is consistently passing. When you change the test to expect a different RadioButton it fails as expected.